includes/class-wcs-importer.php

```diff
@@ -330,16 +330,29 @@ public static function import_subscription( $data ) {
 			}
 
 			switch ( $date_type ) {
+				case 'cancelled_date':
+					if ( 'cancelled_date' === $date_type && ! in_array( $status, wcs_get_subscription_ended_statuses() ) ) {
+						$result['error'][] = sprintf( __( 'Cannot set a %s date for an active subscription.', 'wcs-import-export' ), $date_type );
+					}
 				case 'end_date' :
-					if ( ! empty( $dates_to_update['next_payment_date'] ) && strtotime( $datetime ) <= strtotime( $dates_to_update['next_payment_date'] ) ) {
+					if ( 'end_date' === $date_type && ! empty( $dates_to_update['next_payment_date'] ) && strtotime( $datetime ) <= strtotime( $dates_to_update['next_payment_date'] ) ) {
 						$result['error'][] = sprintf( __( 'The %s date must occur after the next payment date.', 'wcs-import-export' ), $date_type );
 					}
+					if ( 'end_date' === $date_type && ! empty( $dates_to_update['cancelled_date'] ) && strtotime( $datetime ) <= strtotime( $dates_to_update['cancelled_date'] ) ) {
+						$result['error'][] = sprintf( __( 'The %s date must occur after the cancelled date.', 'wcs-import-export' ), $date_type );
+					}
+
+					if ( 'end_date' === $date_type && empty( $dates_to_update['cancelled_date'] ) ) {
+						$dates_to_update['cancelled_date'] = $datetime;
+						ksort( $dates_to_update ); // Sort so that `cancelled_date` is processed before `end_date`, otherwise updating dates will trigger a fatal.
+						$result['warning'][] = sprintf( __( 'Setting %1$s date requires a cancelled date. Setting the cancelled date to %2$s.', 'wcs-import-export' ), $date_type, $datetime );
+					}
```

Comment:
I just ran a test and got this error. I think there's a couple of things wrong here. 

Cancelled dates represent the date the customer or merchant cancelled a subscription. 
The end date is when a subscription will terminate (expire) for ever. 

If you purchase a subscription with a defined length (ie predetermined automatic expirataion), when the customer signs up, it will have an end date. So subscriptions with an end date don't require cancelled dates.

Reply:
You're right about this, too. When testing I kept running into an error message when trying to set an `end_date` on a subscription with `wc-cancelled` status that didn't already have a `cancelled_date`. However, this turns out to be a red herring which I'll explain below.